### ISSUE TITLE
Infinite loop on ":u" string

### ISSUE THREAD

**Issue by daleholborow** [URL-Detector/URL-Detector#5]
Have found a bug in the parsing logic whereby the UrlDetector.readDefault logic gets stuck in an infinite loop and will crash your application in such simple cases as where a string contains a colon followed by an unexpected character combination such that it tests as NOT a username:password pair. I have a simple workaround that I will publish to my C# port (https://github.com/eladaus/URL-Detector) but I thought i would report the issue to the Java community as well and you can implement my patch or find an alternative solution, as suites you.

To repro, run this test:

```java
    @Test
    public void testDalesKillerString3() {
        var sourceText = " :u "; // kills loop in UrlDetector.readDefault()
        var flags = UrlDetectorOptions.ALLOW_SINGLE_LEVEL_DOMAIN;
        var links = new UrlDetector(sourceText, flags)
            // BEWARE THE INFINITE LOOP THAT HAPPENS HERE:
            .detect();
        Assert.assertEquals(0, links.size());
    }

**Comment by cohendekel** [comment]
Thanks for the heads up 👍

**Comment by daleholborow** [comment]
C# logic updated now in my 1.3.0 release. Please check there , in particular the UrlDetector.cs file, and of course, if you find any issues in the implementation or can think of a better solution, please let me know!

**Comment by pgalbraith** [comment]
Thanks for the report, Dale.  I added a quick fix in the main loop to just throw away the next character in the stream if no apparent progress has been made, I hope it's really that simple!

The fix is in the develop branch and I've pushed a 0.1.22-SNAPSHOT build to the Sonatype repo.

**Comment by pgalbraith** [comment]
By the way, @daleholborow if you have any interest in adding your port to this overarching URL-Detector organization, please drop me line.

**Comment by daleholborow** [comment]
@pgalbraith I sent you an email, let me know what I should do :)

**Comment by pgalbraith** [comment]
@daleholborow that's odd I don't think I got an email, I just double checked my spam folder too, but regardless I'll take this to mean that you might be interested in moving your port here.  I'll add you as a member and you can then create a new repo under this group, if you're so inclined!

**Comment by pgalbraith** [comment]
Fixed in 0.1.22
